XenAdmin/Diagnostics/Checks/AssertCanEvacuateCheck.cs

```diff
@@ -184,6 +184,22 @@ private Problem GetProblem(IXenConnection connection, XenRef<VM> vmRef, string[]
 
                         return new NoPVDrivers(this, vm);
 
+                    case Failure.VM_LACKS_FEATURE:
+                        vm = connection.Resolve<VM>(vmRef);

```

Comment:
I see what you're saying regarding the default case, but I think resolving should be the first check because is it meaningful to have a VM problem without a resolved VM?
Beyond this, I think this method is somewhat inefficient because it throws exceptions only to catch, log, and rethrow them, and it could be written in a better way. Is this is a general trunk quality bug? If so, we could do some refactoring.